deploy/deploy-api.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+#
+# Deploy API Server on GCP.
+
+if (( $# != 1 ))
+then
+  echo "Usage: deploy/deploy-api.sh <dataset>"
+  echo "  where <dataset> is the name of a directory in api/dataset_config/"
+  echo "Run this script from project root"
+  exit 1
+fi
+
+dataset=$1
+project_id=$(grep -v "//" api/dataset_config/${dataset}/deploy.json | jq --raw-output '.project_id')
```

Comment:
This line prints: `grep: api/dataset_config/platinum_genomes/deploy.json: No such file or directory`

I'm not sure what `deploy.json` is, what's this line doing?

Reply:
deploy.json looks like:
```
{
  // GCP project to deploy to
  "project_id": "test-data-explorer"
}
```
After this PR is in, I will add to it dataset configs in data-explorer-indexer, so in the future, when user runs this script, it will exist

Reply:
That's fine, in general it would be nice to order these PRs such that they aren't reliant on future changes.

Reply:
FYI - I wanted to do this PR first, since this PR establishes the automation, in case that changed something about deploy.json. (Note in the PR description, I said I would add deploy.json to data-explorer-indexers repro after this PR is in.)